### PR TITLE
included new type_uuid

### PR DESCRIPTION
**Pull request by alvarosimon**
UUID type is used by different Quattor components. It makes sense to include it into pan/types

### CODE REVIEW

**Comment by alvarosimon** [comment]
@stdweird already told me that we must change the old comments code to the new documentation format, I will do that (that's the reason why I don't like to modify old core... :smile:)

**Comment by jrha** [comment]
:grinning: Hey, it's just an opportunity to make things better right...?

**Comment by alvarosimon** [comment]
@jrha @stdweird  sure! :+1:  I have changed the comments to use the new pan annotations, please let me know if this is ok...

File: pan/types.pan [hunk 1]
```diff
@@ -1,53 +1,50 @@
...
-#     * email
-#     * network_name
-#
-################################################################################
+@documentation{
+This is 'namespaces/standard/pan/types.tpl', a pan-templates's file
+
+VERSION:    3.2.7, 21/08/09 22:22
+AUTHOR:     Martin Bock
+MAINTAINER: Marco Emilio Poleggi <Marco.Emilio.Poleggi@cern.ch>, German Cancio <German.Cancio.Melia@cern.ch>, Michel Jouvin <jouvin@lal.in2p3.fr>
```
Comment (jrha) [inline pan/types.pan]: Should these be in `@maintainer{}` blocks?

I'm not sure, just looking at the examples in https://quattor-pan.readthedocs.org/en/pan-10.2/pan-book/pan-book.html#annotations...

Comment (alvarosimon) [inline pan/types.pan]: I think so, I was just wondering about that as well.. I will include a `@maintainer` block for each one

File: pan/types.pan [hunk 2]
```diff
@@ -1,53 +1,61 @@
...
+@maintainer{
+  name = German Cancio
+  email = German.Cancio.Melia@cern.ch
+}
+@maintainer{
+  name = Michel Jouvin
+  email = jouvin@lal.in2p3.fr
+}
+@documentation{
+This is 'namespaces/standard/pan/types.tpl', a pan-templates's file
```
Comment (stdweird) [inline pan/types.pan]: remove

Comment (stdweird) [inline pan/types.pan]: remove whole block till the actual documentation (`Data type ...`). This is irrelevant, ancient and redundant

File: pan/types.pan [hunk 3]
```diff
@@ -1,53 +1,61 @@
...
-#     * port
-#     * hostport
-#     * URI
-#     * absoluteURI
-#     * hostURI
-#     * email
-#     * network_name
-#
-################################################################################
+@maintainer{
```
Comment (stdweird) [inline pan/types.pan]: really, in what sense of `maintainer` would that be? they are authors for sure, but maintainers? (except @jouvin ofcourse).

File: pan/types.pan [hunk 4]
```diff
@@ -1,53 +1,61 @@
...
-#
-# Ex: 20040825120123Z
-#     20040825120123+0100
-#     20040825120123,5
-#     20040825120123.5
-#     20040825120123.5-0123
-###################################################################
+
+@documentation{
+function is_asndate.
```
Comment (stdweird) [inline pan/types.pan]: remove this and the `AUTHOR` field. this is not relevant and very likely outdated

File: pan/types.pan [hunk 5]
```diff
@@ -148,37 +156,37 @@ function is_asndate = {
 };
 
 
-###################################################################
-# type_asndate
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: simply remove this (it contains no actual added anything; except for the author field, but that is not the best way to keep this around.)

File: pan/types.pan [hunk 6]
```diff
@@ -148,37 +156,37 @@ function is_asndate = {
...
-# optional.
-#
-# Comparison function not yet implemented.
-#
-# Ex: 2004-08-25
-#     2004-08-25T12:32:00.01+05:00
-#     2004-08-25T12:32:00Z
-###################################################################
+@documentation{
+function is_isodate
```
Comment (stdweird) [inline pan/types.pan]: same remarks: don't repeat the funciton name, and remove the author part

File: pan/types.pan [hunk 7]
```diff
@@ -275,30 +283,30 @@ function is_isodate = {
 };
 
 
-###################################################################
-# type_isodate
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove

File: pan/types.pan [hunk 8]
```diff
@@ -275,30 +283,30 @@ function is_isodate = {
...
-# and separated with a colon or a hyphen.  Within a value you must
-# use a consistent separator.
-#
-# Ex.: 00:D0:59:33:F6:30
-#      00-D0-59-33-F6-30
-#
-# Both upper and lower-case hex digits are accepted.
-###################################################################
+@documentation{
+function is_hwaddr
```
Comment (stdweird) [inline pan/types.pan]: remove function and AUTHOR

File: pan/types.pan [hunk 9]
```diff
@@ -309,23 +317,23 @@ function is_hwaddr = {
 };
 
 
-###################################################################
-# type_hwaddr
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove completely

File: pan/types.pan [hunk 10]
```diff
@@ -309,23 +317,23 @@ function is_hwaddr = {
...
 
-###################################################################
-# function is_ipv4
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-#
-# Function to validate an IPv4 address in dotted-decimal notation.
-###################################################################
+@documentation{
+function is_ipv4
```
Comment (stdweird) [inline pan/types.pan]: remove function and author

File: pan/types.pan [hunk 11]
```diff
@@ -356,25 +364,25 @@ function is_ipv4 = {
 };
 
 
-###################################################################
-# type_ipv4
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove whole block

File: pan/types.pan [hunk 12]
```diff
@@ -356,25 +364,25 @@ function is_ipv4 = {
...
 
-###################################################################
-# function is_ipv6
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-#
-# Function to validate an IPv6 address.
-###################################################################
+@documentation{
+function is_ipv6
```
Comment (stdweird) [inline pan/types.pan]: remove function/author

File: pan/types.pan [hunk 13]
```diff
@@ -384,10 +392,15 @@ function is_ipv6 = {
     return(is_ipv6_full(ip) || is_ipv6_short(ip));
 };
 
-# Check that argument is a valid full IPv6 address.  This is
-# eight, 16-bit numbers represented in hexadecimal notation and
-# separated by colons.  Leading zeros of each field can be
-# suppressed.
+
+@documentation{
+function is_ipv6_full
```
Comment (stdweird) [inline pan/types.pan]: remove function

File: pan/types.pan [hunk 14]
```diff
@@ -399,9 +412,14 @@ function is_ipv6_full = {
     return(match(ip,'^[\dA-Fa-f]{1,4}(:[\dA-Fa-f]{1,4}){7}$'));
 };
 
-# Check that argument is a valid short IPv6 address.  The shortened
-# form of the IPv6 address allows a single instance of a double
-# colon (::) to replace any number of contiguous zero fields.
+
+@documentation{
+function is_ipv6_short
```
Comment (stdweird) [inline pan/types.pan]: remove function

File: pan/types.pan [hunk 15]
```diff
@@ -432,23 +450,23 @@ function is_ipv6_short = {
 };
 
 
-###################################################################
-# type_ipv6
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove

File: pan/types.pan [hunk 16]
```diff
@@ -432,23 +450,23 @@ function is_ipv6_short = {
...
 
-###################################################################
-# function is_ip
-#
-# AUTHOR: Rafael A. Garcia Leiva <angel.leiva@uam.es>
-#
-# Checks that the given address is a valid IPv4 or IPv6 address.
-###################################################################
+@documentation{
+function is_ip
```
Comment (stdweird) [inline pan/types.pan]: remove function/author

File: pan/types.pan [hunk 17]
```diff
@@ -457,30 +475,29 @@ function is_ip = {
 };
 
 
-###################################################################
-# type type_ip
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove

File: pan/types.pan [hunk 18]
```diff
@@ -457,30 +475,29 @@ function is_ip = {
...
-# of the domain name is separated by a period.  The individual parts
-# must begin with a letter or digit, end with a letter or digit, 
-# and may contain letters, digits, or hyphens in the middle.
-#
-# The relevant RFC's for host name syntax are 952, 1053, and 1123
-# (section 2.1).
-#
-###################################################################
+@documentation{
+function is_fqdn
```
Comment (stdweird) [inline pan/types.pan]: remove function/author

File: pan/types.pan [hunk 19]
```diff
@@ -490,23 +507,23 @@ function is_fqdn = {
 };
 
 
-###################################################################
-# type_fqdn
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove

File: pan/types.pan [hunk 20]
```diff
@@ -490,23 +507,23 @@ function is_fqdn = {
...
 
-###################################################################
-# function is_hostname
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-#
-# Validates a hostname to be either ip or fqdn.
-###################################################################
+@documentation{
+function is_hostname
```
Comment (stdweird) [inline pan/types.pan]: remove function/author

File: pan/types.pan [hunk 21]
```diff
@@ -516,24 +533,24 @@ function is_hostname = {
 };
 
 
-###################################################################
-# type_hostname
-#
-# AUTHOR: Charles Loomis <charles.loomis@cern.ch>
-###################################################################
+@documentation{
```
Comment (stdweird) [inline pan/types.pan]: remove

**Comment by stdweird** [comment]
@alvarosimon no need to repeat the function or type is is annotated; that is covered by the tools that generate the docs from the annotations.
wrt the author, i'd remove it, since it's not adding much (and we don't do it anywhere else).

@jrha @jouvin do we have a good way to credit the authors?

File: pan/types.pan (hunk 3 above)
Comment (alvarosimon) [inline pan/types.pan]: @stdweird I have the same question, I only see `maintainer` block from pan book, and probably this info is outdated what we do? we keep just @jouvin as maintainer?..

**Comment by alvarosimon** [comment]
@stdweird ok I will remove the unnecessary documentation blocks

File: pan/types.pan (hunk 3 above)
Comment (jrha) [inline pan/types.pan]: This seems related to quattor/maven-tools#16, perhaps we should just have just have `@contributor`s and leave it at that.

**Comment by alvarosimon** [comment]
@stdweird @jrha ok I have replaced `maintainer` by `contributor`, I think that now is more clean...

**Pull request merged**